airflow/hooks/webhdfs_hook.py

```diff
@@ -42,70 +41,91 @@ class AirflowWebHDFSHookException(AirflowException):
+                self.log.debug('Read operation on namenode %s failed with error: %s',
+                               connection.host, hdfs_error)
+
+        hosts = [connection.host for connection in connections]
+        error_message = 'Read operations failed on the namenodes below:\n{hosts}'.format(
+            hosts='\n'.join(hosts))
+        raise AirflowWebHDFSHookException(error_message)
+
+    def _get_client(self, connection):
+        connection_str = 'http://{host}:{port}'.format(host=connection.host, port=connection.port)
+
+        if _kerberos_security_mode:
+            client = KerberosClient(connection_str)
+        else:
+            proxy_user = self.proxy_user or connection.login
+            client = InsecureClient(connection_str, user=proxy_user)
+
+        return client
 
     def check_for_path(self, hdfs_path):
         """
         Check for the existence of a path in HDFS by querying FileStatus.
+
+        :param hdfs_path: The path to check.
+        :type hdfs_path: str
+        :return: True if the path exists and False if not.
+        :rtype: bool
         """
-        c = self.get_conn()
-        return bool(c.status(hdfs_path, strict=False))
+        conn = self.get_conn()
+
+        status = conn.status(hdfs_path, strict=False)
+        return bool(status)
 
-    def load_file(self, source, destination, overwrite=True, parallelism=1,
-                  **kwargs):
+    def load_file(self, source, destination, overwrite=True, parallelism=1, **kwargs):
         r"""
-        Uploads a file to HDFS
+        Uploads a file to HDFS.
 
-        :param source: Local path to file or folder. If a folder, all the files
-          inside of it will be uploaded (note that this implies that folders empty
-          of files will not be created remotely).
+        :param source: Local path to file or folder.
+            If it's a folder, all the files inside of it will be uploaded.
+            .. note:: That this implies that folders empty of files will not be created remotely.
         :type source: str
-        :param destination: PTarget HDFS path. If it already exists and is a
-          directory, files will be uploaded inside.
+        :param destination: PTarget HDFS path.
+            If it already exists and is a directory, files will be uploaded inside.
         :type destination: str
         :param overwrite: Overwrite any existing file or directory.
         :type overwrite: bool
-        :param parallelism: Number of threads to use for parallelization. A value of
-          `0` (or negative) uses as many threads as there are files.
+        :param parallelism: Number of threads to use for parallelization.
+            A value of `0` (or negative) uses as many threads as there are files.
         :type parallelism: int
-        :param \*\*kwargs: Keyword arguments forwarded to :meth:`upload`.
-
-
+        :param \**kwargs: Keyword arguments forwarded to :meth:`upload`.
```

Comment:
Can you use the full method path - `hdfs.client.Client.upload`? This will allow us to automatically link to external documentation using sphinx.ext.intersphinx.
You will also need to add the mapping in the variable `intersphinx_mapping` in `docs/conf.py`.
```
https://hdfscli.readthedocs.io/en/latest/objects.inv
```
If you would like to see a list of documentation elements, then you should execute a command 
```
python -msphinx.ext.intersphinx https://hdfscli.readthedocs.io/en/latest/objects.inv
```
What do you think about this improvement? 
Reference:
http://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html

Reply:
I like it 👍